com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/resource/persistence/DirectLinkingEObjectInputStream.java

```diff
@@ -67,8 +67,8 @@ public EObject readEObject(final Resource context) throws IOException {
             eObject = (EObject) eObject.eGet(feature, false);
           }
         } else {
-          LOGGER.error(String.format("Failed to get EStructuralFeature for Eclass : %s, at readCompressedInt : %d", eObject.eClass(), next));
-
+          LOGGER.info(String.format("Failed to get EStructuralFeature for Eclass : %s, at readCompressedInt : %d", eObject.eClass(), next));
+          throw new IllegalStateException("feature is null"); //$NON-NLS-1$

```

Comment:
what about removing the log and adding

throw new NullPointerException(String.format("EStructuralFeature can't be null for Eclass : %s, at readCompressedInt : %d", eObject.eClass(), next))